extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java

```diff
@@ -53,10 +56,31 @@ public VertxSpringCloudConfigGateway(SpringCloudConfigClientConfig springCloudCo
             throw new IllegalArgumentException("Value: '" + springCloudConfigClientConfig.url
                     + "' of property 'quarkus.spring-cloud-config.url' is invalid", e);
         }
-        this.vertx = Vertx.vertx();
+        this.vertx = createVertxInstance();
         this.webClient = createHttpClient(vertx, springCloudConfigClientConfig, tlsConfig);
     }
 
+    private Vertx createVertxInstance() {
+        // We must disable the async DNS resolver as it can cause issues when resolving the Vault instance.
+        // This is done using the DISABLE_DNS_RESOLVER_PROP_NAME system property.
+        // The DNS resolver used by vert.x is configured during the (synchronous) initialization.
+        // So, we just need to disable the async resolver around the Vert.x instance creation.
+        String originalValue = System.getProperty(DISABLE_DNS_RESOLVER_PROP_NAME);
```

Comment:
If it was set on the command line with `-Dvertx.disableDnsResolver` then, I think, the value will be null and it will be cleared in the finally.